lib/puppet/util.rb

```diff
@@ -348,6 +348,18 @@ def uri_to_path(uri)
   end
   module_function :uri_to_path
 
+  # Takes a string containing characters that should be encoded in a Uri
+  # The output is safe to pass to URI.parse or URI::Generic.build  because it
+  # unescapes characters / and : prior to producing a value
+  def uri_escape(path)
+    return unless path
```

Comment:
This is a slight API change in that `URI.escape` and `CGI.escape` fail with a `nil` parameter, but our `uri_escape` does not. Does that matter?